File: frontend/types/analysis.ts
// Type definitions matching the backend API response

export interface AnalysisResult {
 repository: string;
 summary: Summary;
 nodes: Node[];
 edges: Edge[];
 message?: string;
}

export interface Summary {
 total_files: number;
 total_functions: number;
 high_risk_files: number;
}

export interface Node {
 id: string;
 type: string;
 metrics: Metrics;
}

export interface Metrics {
 complexity: number;
 functions_count: number;
 risk_level: 'low' | 'medium' | 'high';
 lines_of_code?: number;
}

export interface Edge {
 source: string;
 target: string;
 type: string;
}

export interface AnalyzeRequest {
 source_url: string;
}

export interface HealthCheckResponse {
 status: string;
 service: string;
 timestamp: string;
 version: string;
}

File: backend/internal/models/analysis.go
package models

// AnalysisResult represents the complete analysis output
type AnalysisResult struct {
 Repository string `json:"repository"`
 Summary Summary `json:"summary"`
 Nodes []Node `json:"nodes"`
 Edges []Edge `json:"edges"`
}

// Summary contains high-level statistics about the codebase
type Summary struct {
 TotalFiles int `json:"total_files"`
 TotalFunctions int `json:"total_functions"`
 HighRiskFiles int `json:"high_risk_files"`
}

// Node represents a file in the codebase
type Node struct {
 ID string `json:"id"`
 Type string `json:"type"`
 Metrics Metrics `json:"metrics"`
}

// Metrics contains analysis metrics for a file
type Metrics struct {
 Complexity int `json:"complexity"`
 FunctionsCount int `json:"functions_count"`
 RiskLevel string `json:"risk_level"` // "low", "medium", "high"
 LinesOfCode int `json:"lines_of_code"`
}

// Edge represents a dependency relationship between files
type Edge struct {
 Source string `json:"source"`
 Target string `json:"target"`
 Type string `json:"type"` // "import", "export", etc.
}

// FileInfo represents parsed information about a single file
type FileInfo struct {
 Path string
 Imports []string
 Exports []string
 Functions []FunctionInfo
 Complexity int
 LinesOfCode int
}

// FunctionInfo represents a function declaration
type FunctionInfo struct {
 Name string
 LineStart int
 LineEnd int
 Complexity int
}